### DISCUSSION TITLE
No serial port

### DISCUSSION THREAD

**Discussion by dwfunk4475** [gizmo-platform/gizmo#35] [General]
For starters, everytime we plug this gizmo's student side Pico in the UF2 drive appears without holding down the bootsel.

We're trying to run the sample code to read the right joy stick. IDE compiles, I see new file show up on UF2 drive, but UF2 does not 'go away' and begin running the program.

Repowering does not run the apparently downloaded program.

Pico never shows up on a COM port and we are unable to connect serial monitor. 

Suggestions?

**Comment by the-maldridge** [comment]
Hmm, this could be a handful of things.  If you push on the bootsel button does it click on the student side?  There should be a satisfying tactile "click" as it moves between states.

If that's not moving smoothly, this might be as simple as the button lid being stuck, you can gently pull up on it to dislodge it.

If it is free to move, there could be something deeper going on.  Can you share what programing language you're using?  I suspect Arduino because you mention compiling, but I want to be sure.  On the very first load from the IDE under certain circumstances you need to change the mode to upload to the code directly using filesystem operations.

**Reply by dwfunk4475** [reply]
Definitely a tactical 'click.' We removed the cover to make sure and to see the onboard LED. (Kinda dumb to cover that LED and then include it in all the examples to demonstrate the program is running!)

Programming environment has been; Arduino IDE, Python, and Blockly. 

Nothing copied to the UF2 drive runs.

We never see this Pico2 show up on a COM port and cannot establish a terminal session. (Monitor, Plotter)

Put headers on one of my own Pico2's and replaced this one. Everything works fine now. Going back to the Hub to get this Pico2 replaced.


-david

**Reply by the-maldridge** [reply]
Hmm, that is indeed an issue.  If your hub contact knows, would you mind telling me where they ordered the pico from?  We're trying to track down if all our problem picos came from the same manufacturing run.
